core/src/weighted_shuffle.rs

```diff
@@ -41,16 +43,18 @@ pub fn weighted_best(weights_and_indicies: &[(u64, usize)], rng: ChaChaRng) -> u
         return 0;
     }
     let total_weight: u64 = weights_and_indicies.iter().map(|x| x.0).sum();
-    let mut best_weight = 0;
+    let mut lowest_weight = std::u128::MAX;
     let mut best_index = 0;
     for v in weights_and_indicies {
+        // This generates an "inverse" weight but it avoids floating point math
         let x = (total_weight / v.0)
             .to_u64()
             .expect("values > u64::max are not supported");
         // capture the u64 into u128s to prevent overflow
-        let weight = (&mut rng).gen_range(1, u128::from(std::u16::MAX)) * u128::from(x);
-        if weight > best_weight {
-            best_weight = weight;
+        let computed_weight = (&mut rng).gen_range(1, u128::from(std::u16::MAX)) * u128::from(x);
```

Comment:
I know the original code has that `(&mut rng)`, but is it possible to reduce that to simply `rng`. That annotation looks redundant.